hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+module CalendarResponse where
+
+import Data.List (sort)
+import Data.List.Split (splitOn)
+import Data.Time (Day, UTCTime, addDays, formatTime, getCurrentTime, utctDay)
+import Happstack.Server (ServerPart, Response, toResponse)
+import Control.Monad.IO.Class (liftIO)
+import System.Locale
+import Config (firstMondayFall, lastWednesdayFall ,firstMondayWinter, lastMondayWinter, outDay)
+import Database.CourseQueries (returnTutorialTimes, returnLectureTimes)
+import qualified Data.Text as T
+import Database.Tables as Tables
+
+-- | Returns a CSV file of events as requested by the user.
+calendarResponse :: String -> ServerPart Response
+calendarResponse courses =
+    liftIO $ getCalendar courses
+
+-- | Gets together all the pieces of the program.
+getCalendar :: String -> IO Response
+getCalendar courses = do
+    let courseInfo = getCoursesInfo courses
+    databaseInfo <- sequence $ map pullDatabase courseInfo
+    currentTime <- getCurrentTime
+    let timeSystem = getTime currentTime
+    let events = concat $ map (getEvent timeSystem) databaseInfo
```

Comment:
If I try to use 'fmap' here I will end up using it three times. I think, it is a little bit clearer the way it is. I will be happy to change it if you think otherwise.